base/pcre.jl

```diff
@@ -25,6 +25,8 @@ end
 
 const THREAD_MATCH_CONTEXTS = Ptr{Cvoid}[C_NULL]
 
+PCRE_COMPILE_LOCK = nothing
```

Comment:
Would it be worth having this as a typed Ref, just as a hint to the reader what will en up there?

Reply:
Oh, I would love to, but threads.jl has not been loaded yet so the type is not available. Regexes turn out to be used in some fundamental places like show.jl and version.jl, so there is unfortunately a circular dependency.